### PR TITLE
Queue depth debugging code

### PR DESCRIPTION
**Pull request by roxanneskelly**
Add code for debugging queue depth to detect starvation issues, queue overuse, etc.

This code is currently enabled, but can be disabled by removing the DEBUG_EVENT_QUEUE define in QtHelpers.h

This is part of BUGZ-128

### CODE REVIEW

File: interface/src/Application.cpp [hunk 1]
```diff
@@ -4075,14 +4046,14 @@ bool Application::event(QEvent* event) {
         case ApplicationEvent::Idle:
             idle();
 
-#ifdef DEBUG_EVENT_QUEUE
+#ifdef DEBUG_EVENT_QUEUE_DEPTH
```
Comment (howard-stearns) [inline interface/src/Application.cpp]: Is this ententionally not def'd by standard builds?

Comment (roxanneskelly) [reply interface/src/Application.cpp]: Yeah, this was old debugging code that was left in there for manual debugging some time ago.  I didn't want it included by default as it spams the event queue when it gets big...and in the wild it does get big enough to cause the spam.

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19405/HighFidelity-Beta-PR15551-4ad3163.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19405/HighFidelity-Quest-Beta-PR15551-4ad3163.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19405/HighFidelity-Beta-PR15551-4ad3163.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19405/HighFidelity-Beta-Interface-PR15551-4ad3163.exe

**Comment by roxanneskelly** [comment]
build me please

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19411/HighFidelity-Beta-PR15551-aa16430.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19411/HighFidelity-Quest-Beta-PR15551-aa16430.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19411/HighFidelity-Beta-PR15551-aa16430.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19411/HighFidelity-Beta-Interface-PR15551-aa16430.exe

**Comment by howard-stearns** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19421/HighFidelity-Beta-PR15551-aa16430.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19421/HighFidelity-Quest-Beta-PR15551-aa16430.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19421/HighFidelity-Beta-PR15551-aa16430.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19421/HighFidelity-Beta-Interface-PR15551-aa16430.exe

**Comment by roxanneskelly** [comment]
build this please

File: libraries/networking/src/DomainHandler.cpp [hunk 2]
```diff
@@ -567,9 +567,19 @@ bool DomainHandler::checkInPacketTimeout() {
...
+
+        auto nodeList = DependencyManager::get<NodeList>();
+
         // we haven't heard back from DS in MAX_SILENT_DOMAIN_SERVER_CHECK_INS
         // so emit our signal that says that
-        qCDebug(networking_ice) << "Limit of silent domain checkins reached";
+
+#ifdef DEBUG_EVENT_QUEUE
+        int nodeListQueueSize = ::hifi::qt::getEventQueueSize(nodeList->thread());
+        qCDebug(networking) << "Limit of silent domain checkins reached (network qt queue: " << nodeListQueueSize << ")";
```
Comment (howard-stearns) [inline libraries/networking/src/DomainHandler.cpp]: I wonder if we need to see the growth? e.g., reporting queue size at each failed checkin, and not just when we reach the limit of silent checkins?

(I don't know what the right thing is. Just speculating.)

Comment (roxanneskelly) [reply libraries/networking/src/DomainHandler.cpp]: more info is better, at least for now, I suspect

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19436/HighFidelity-Beta-PR15551-68fb43e.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19436/HighFidelity-Beta-Interface-PR15551-68fb43e.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19436/HighFidelity-Beta-PR15551-68fb43e.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19436/HighFidelity-Beta-Interface-PR15551-68fb43e.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19436/HighFidelity-Beta-PR15551-68fb43e.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19436/HighFidelity-Quest-Beta-PR15551-68fb43e.apk)

**Comment by howard-stearns** [comment]
being tested by the team

File: assignment-client/src/audio/AudioMixerSlavePool.cpp [hunk 3]
```diff
@@ -113,6 +113,19 @@ void AudioMixerSlavePool::each(std::function<void(AudioMixerSlave& slave)> funct
     }
 }
 
+#ifdef DEBUG_EVENT_QUEUE
+void AudioMixerSlavePool::queueStats(QJsonObject & stats) {
```
Comment (hyperlogic) [inline assignment-client/src/audio/AudioMixerSlavePool.cpp]: https://github.com/highfidelity/hifi/blob/master/CODING_STANDARD.md#325-c-pointers-and-references-should-have-their-reference-symbol-next-to-the-type-rather-than-to-the-name

File: assignment-client/src/audio/AudioMixerSlavePool.h [hunk 4]
```diff
@@ -72,6 +72,10 @@ class AudioMixerSlavePool {
     // iterate over all slaves
     void each(std::function<void(AudioMixerSlave& slave)> functor);
 
+#ifdef DEBUG_EVENT_QUEUE
+    void queueStats(QJsonObject & stats);
```
Comment (hyperlogic) [inline assignment-client/src/audio/AudioMixerSlavePool.h]: https://github.com/highfidelity/hifi/blob/master/CODING_STANDARD.md#325-c-pointers-and-references-should-have-their-reference-symbol-next-to-the-type-rather-than-to-the-name

File: assignment-client/src/avatars/AvatarMixerSlavePool.cpp [hunk 5]
```diff
@@ -117,6 +117,19 @@ void AvatarMixerSlavePool::each(std::function<void(AvatarMixerSlave& slave)> fun
     }
 }
 
+#ifdef DEBUG_EVENT_QUEUE
+void AvatarMixerSlavePool::queueStats(QJsonObject & stats) {
```
Comment (hyperlogic) [inline assignment-client/src/avatars/AvatarMixerSlavePool.cpp]: https://github.com/highfidelity/hifi/blob/master/CODING_STANDARD.md#325-c-pointers-and-references-should-have-their-reference-symbol-next-to-the-type-rather-than-to-the-name

File: interface/resources/qml/Stats.qml [hunk 6]
```diff
@@ -147,6 +147,16 @@ Item {
                                     "Parabolas:\t" + root.parabolaPicksUpdated.x + "/" + root.parabolaPicksUpdated.y + "/" + root.parabolaPicksUpdated.z + "\n    " +
                                     "Colliders:\t" + root.collisionPicksUpdated.x + "/" + root.collisionPicksUpdated.y + "/" + root.collisionPicksUpdated.z
                     }
+                    StatText {
+                        visible: { root.eventQueueDebuggingOn && root.expanded }
+                        text: { if (root.eventQueueDebuggingOn)
```
Comment (hyperlogic) [inline interface/resources/qml/Stats.qml]: https://github.com/highfidelity/hifi/blob/master/CODING_STANDARD.md#4111-single-statement-if-else-for-or-while-statements-must-be-written-with-braces

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19460/HighFidelity-Beta-PR15551-a759549.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19460/HighFidelity-Quest-Beta-PR15551-a759549.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19460/HighFidelity-Beta-PR15551-a759549.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19460/HighFidelity-Beta-Interface-PR15551-a759549.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19460/HighFidelity-Beta-PR15551-a759549.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19460/HighFidelity-Beta-Interface-PR15551-a759549.exe

**Pull request merged**